src/auto_scheduler/measure_record.cc

```diff
@@ -183,7 +186,12 @@ struct Handler<::tvm::auto_scheduler::SearchTaskNode> {
       reader->Read(hardware_params_node.get());
       s = reader->NextArrayItem();
       data->hardware_params = ::tvm::auto_scheduler::HardwareParams(hardware_params_node);
-      ICHECK(!s);
+      if (s) {
+        reader->Read(&str_value);
+        data->target_host = ::tvm::Target(str_value);
+        s = reader->NextArrayItem();
+        ICHECK(!s);
```

Comment:
This check should be out of this if-statement.

Reply:
Move out will break back compatibility